internal/clients/sources/sources_impl.go

```diff
@@ -2,12 +2,54 @@ package sources
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"io"
 
 	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
 )
 
+type AppType struct {
+	Id          string `json:"id"`
+	Name        string `json:"name"`
+	DisplayName string `json:"display_name"`
+}
+
+type dataElement struct {
+	Data []AppType `json:"data"`
+}
+
 func getSourcesClient(ctx context.Context) (SourcesIntegration, error) {
-	return NewClientWithResponses(config.Sources.URL)
+	client, err := NewClientWithResponses(config.Sources.URL)
+	if err != nil {
+		return client, err
+	}
+	if config.Sources.AppTypeId == "" {
+		ctxval.GetLogger(ctx).Info().Msg("Fetching the Application Type ID of Provisioning for Sources")
+		resp, err := client.ListApplicationTypes(ctx, &ListApplicationTypesParams{})
+		if err != nil {
+			ctxval.GetLogger(ctx).Warn().Msgf("Failed to fetch ApplicationTypes from sources: %w", err)
```

Comment:
I think you need to add it as an error in sources_error.go 
and to use errors.New to fix that error.